go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -179,51 +253,70 @@ func createWorkingSetForLocalBranch(ctx *sql.Context, ddb *doltdb.DoltDB, branch
 	if err != nil {
-		return errors.New("fatal: unable to read from data repository")
+		return "", errors.New("fatal: unable to read from data repository")
 	}
 
 	if len(remoteRefs) == 0 {
-		return fmt.Errorf("error: could not find %s", branchName)
+		if doltdb.IsValidCommitHash(branchName) && apr.Contains(cli.MoveFlag) {
+
+			// User tried to enter a detached head state, which we don't support.
+			// Inform and suggest that they check-out a new branch at this commit instead.
+
+			return "", fmt.Errorf("dolt does not support a detached head state. To create a branch at this commit instead, run:\n\n"+
+				"\tdolt checkout %s -b {new_branch_name}\n", branchName)
+		}
+		return "", fmt.Errorf("error: could not find %s", branchName)
 	} else if len(remoteRefs) == 1 {
 		remoteRef := remoteRefs[0]
 		err = actions.CreateBranchWithStartPt(ctx, dbData, branchName, remoteRef.String(), false, rsc)
 		if err != nil {
-			return err
+			return "", err
 		}
 
 		// We need to commit the transaction here or else the branch we just created isn't visible to the current transaction,
 		// and we are about to switch to it. So set the new branch head for the new transaction, then commit this one
 		sess := dsess.DSessFromSess(ctx.Session)
 		err = commitTransaction(ctx, sess, rsc)
 		if err != nil {
-			return err
+			return "", err
 		}
 
-		err = checkoutBranch(ctx, dbName, branchName)
+		err = checkoutBranch(ctx, dbName, branchName, apr)
 		if err != nil {
-			return err
+			return "", err
+		}
+
+		// After checking out a new branch, we need to reload the database.
+		dbData, ok := dSess.GetDbData(ctx, dbName)
+		if !ok {
+			return "", fmt.Errorf("Could not reload database %s", dbName)
 		}
 
 		refSpec, err := ref.ParseRefSpecForRemote(remoteRef.GetRemote(), remoteRef.GetBranch())
 		if err != nil {
-			return errhand.BuildDError(fmt.Errorf("%w: '%s'", err, remoteRef.GetRemote()).Error()).Build()
+			return "", errhand.BuildDError(fmt.Errorf("%w: '%s'", err, remoteRef.GetRemote()).Error()).Build()
 		}
 
 		headRef, err := dbData.Rsr.CWBHeadRef()
 		if err != nil {
-			return err
+			return "", err
 		}
 
-		return env.SetRemoteUpstreamForRefSpec(dbData.Rsw, refSpec, remoteRef.GetRemote(), headRef)
+		err = env.SetRemoteUpstreamForRefSpec(dbData.Rsw, refSpec, remoteRef.GetRemote(), headRef)
+		if err != nil {
+			return "", err
+		}
+
+		return remoteRef.GetPath(), nil
 	} else {
-		return fmt.Errorf("'%s' matched multiple (%v) remote tracking branches", branchName, len(remoteRefs))
+		return "", fmt.Errorf("'%s' matched multiple (%v) remote tracking branches", branchName, len(remoteRefs))
 	}
 }
 
-func checkoutNewBranch(ctx *sql.Context, dbName string, dbData env.DbData, apr *argparser.ArgParseResults, rsc *doltdb.ReplicationStatusController) error {
+func checkoutNewBranch(ctx *sql.Context, dbName string, dbData env.DbData, apr *argparser.ArgParseResults, rsc *doltdb.ReplicationStatusController, isGlobal bool) (string, string, error) {
```

Comment:
Input - isGlobal -> isMove

Output - You added two new strings? What are they? gDocs help, but just naming the output variables goes a long way too.

Reply:
Added both docstring and variable names.